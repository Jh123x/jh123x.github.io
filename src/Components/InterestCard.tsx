"use client";
import * as React from "react";
import styled from "styled-components";
import { InterestDescription } from "./types";
import Image from "next/image";
import Link from "next/link";

const InterestCard = ({
  image,
  title,
  paragraph,
  href,
}: InterestDescription) => (
  <InterestCardStyled>
    <Link href={href} target="_blank" rel="noreferrer" aria-label={title}>
      <div className="container">
        <Image src={image.src} width={100} height={100} alt={title} />
        <h4>{title}</h4>
        <p>{paragraph}</p>
      </div>
    </Link>
  </InterestCardStyled>
);

const InterestCardStyled = styled.div`
  background-color: var(--background-dark-grey);
  border-left: 1px solid var(--border-color);
  border-top: 8px solid var(--border-color);
  border-right: 1px solid var(--border-color);
  border-bottom: 1px solid var(--border-color);
  transition: all 0.4s ease-in-out;
  &:hover {
    border-top: 8px solid var(--primary-color);
    transform: translateY(3px);
  }
  .container {
    padding: 1.2rem;
    h4 {
      color: var(--white-color);
      font-size: 1.6rem;
      padding: 1rem 0;
      position: relative;
      &::after {
        content: "";
        width: 4rem;
        background-color: var(--border-color);
        height: 3px;
        position: absolute;
        left: 0;
        bottom: 0;
        border-radius: 10px;
      }
    }

    p {
      padding: 0.8rem 0;
    }
  }
`;

export default InterestCard;
