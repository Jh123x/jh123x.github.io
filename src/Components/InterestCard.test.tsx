import * as React from "react";
import { describe, expect, test } from "@jest/globals";
import InterestCard from "./InterestCard";
import gamedev from "../img/game-dev.svg";

describe("InterestCard", () => {
  test("should render correctly", () => {
    expect(
      <InterestCard
        image={gamedev}
        title="School Blog"
        href="https://nus.jh123x.com"
      />
    ).toMatchSnapshot();
  });
});
