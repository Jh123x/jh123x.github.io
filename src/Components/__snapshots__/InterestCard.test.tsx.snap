// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`InterestCard should render correctly 1`] = `
<InterestCard
  href="https://nus.jh123x.com"
  image="game-dev.svg"
  title="School Blog"
/>
`;
